### PR TITLE
Todo-10-BE: As a user, I want to see all users list

### PR DESCRIPTION
**Pull request by itc02**

### CODE REVIEW

File: app/services/paginate_items_service.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class PaginateItemsService < ActiveInteraction::Base
```
Comment (Deathlive) [inline app/services/paginate_items_service.rb:1]: @itc02 
Es servicy es xorhurd k tam chinjis amenin hti service kiris verin mechan Datan gets anum.
Qez prosto 2 GET servica petq inan usernerin ev todonerin, very ogtagorteluvs **indexum**.

File: app/services/paginate_items_service.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  integer :per, default: nil
+  integer :page, default: nil
+  boolean :paginate, default: false
```
Comment (Deathlive) [inline app/services/paginate_items_service.rb:4]: **per** and **page** chotki tams 5, 10 uranq karoxn `nil` inin ?
K aracharkeyi srazu `items.page(page).per(per)` ogtagortis, avelord cheker petkchi.

File: config/routes.rb [hunk 3]
```diff
@@ -1,11 +1,14 @@
...
     collection do
       delete :delete_all, to: 'todos#delete_all'
-      get :get_owners, to: 'todos#get_owners'
     end
   end
   delete 'users', to: 'users#destroy'
-  resources :users, only: [:index, :create]
+  resources :users, only: [:index, :create] do
+    collection do
+      get :all, to: 'users#get_all_users'
```
Comment (Deathlive) [inline config/routes.rb:11]: `get_all_users` petqa ?

`index` -y gidum ta lox casery-in hti bol kini..

File: app/controllers/todos_controller.rb [hunk 4]
```diff
@@ -1,25 +1,51 @@
...
+      :per => params[:per],
+      :page => params[:page],
+      :items => todo_list_joined_with_users
+    )
+
+    render :json => { 
+      :todos => todos,
+      :total_record_count => TodoList.count
+    }
+
```
Comment (Deathlive) [inline app/controllers/todos_controller.rb:13]: orinak `GetTodosService`
very veradardznuma srazu.
```
{
  todos: todos_data,
  total_record_count: todos_data.count
}
```

is `indexumy` prosto stugums

```
service = GetTodosService.run(...)

if service.valid?
  render json: service.result
else
  render json: service.errors, status: 400
end
```

1) Hinch petqa servicum srazu veradardzru, controllerum starachil  DB zapros anumel.
2) Ete cod-y shata erkar, apa texapoxe Service, ete che tox mna controllerum.

File: app/controllers/todos_controller.rb [hunk 5]
```diff
@@ -1,25 +1,51 @@
...
-    render :json => get_unpaginated_todos
+    render :json => {
+      :todos => PaginateItemsService.run!(
+        :paginate => true,
+        :per => params[:per],
+        :page => params[:page],
+        :items => todo_list_joined_with_users
+      ),
+      :total_record_count => TodoList.count
+    }
```
Comment (Deathlive) [inline app/controllers/todos_controller.rb:48]: destroy po ideye 1-hat chi chinjum ?
Ete tiya piti chi uran `id` tanq ayl voch te `ids`

Mingel destroyich heto xeynq hinch vor pen veradardznum ?
Aveli lyav chi sti aninq ?
```
def destroy
  todo = TodoList.find(params[:id])

    if todo.destroy
      head :no_content
    else
      render json: { errors: todo.errors }, status: 400
    end
end
```

Comment (itc02) [reply app/controllers/todos_controller.rb:48]: Usery kare mi qani hat todo yntre checkboxav, ev 'Delete' kknopkan hop ta. Bidi miangamic lox yntracnen chinchvin. 
Ver mi qani hat chinchvi todo listan, bidi paginationy pahve. Ete veshmin pen serveric xarvevech, usery bidi refresh ane ejy, ver uran koxmic chinchvac todonen tsnavech.

File: config/routes.rb (hunk 3 above)
Comment (itc02) [reply config/routes.rb:11]: Ete `get_all_users`-y inivech, es `axios`-av hunc bidi `index`-in haskacnim, ver lox usernerin veradardzne, vech tvyal `page`-in ev `per`-in hamapatasxanox?

File: app/controllers/todos_controller.rb (hunk 5 above)
Comment (Deathlive) [reply app/controllers/todos_controller.rb:48]: Pervi momenty haskachi.

Baich htek pagination-i pahy che.
Karelchnq pa munq frontum `DELETE` zaprosa aninq `then` ete lox lava, mingel `GET INDEX` zapros kanchinq ?

File: config/routes.rb (hunk 3 above)
Comment (Deathlive) [reply config/routes.rb:11]: Prosto haskanumchum, xeya qez All Usernen petq ete Pagination onis.

Lav isk ete petqa apa prosto karis mihat check xarkis, orinak `witohut_pagination` kam `only_all_users` ev ete `true` apa servisich veradardznums lox aranch paginationi.

File: app/controllers/todos_controller.rb (hunk 5 above)
Comment (itc02) [reply app/controllers/todos_controller.rb:48]: Ha, ti el karoxnq aninq. Dyuza

File: config/routes.rb (hunk 3 above)
Comment (itc02) [reply config/routes.rb:11]: 👍

File: app/services/paginate_items_service.rb (hunk 1 above)
Comment (itc02) [reply app/services/paginate_items_service.rb:1]: Es servicy dra heti-um kiral, hancu qani het pagination petq kya anim, mi hat yndhanur service ini, veric karim ogtvim. Hancu amen donum nuyn peny chi kirim.

Comment (itc02) [reply app/services/paginate_items_service.rb:1]: Vsyo taki chinchim?

**Review by arpinehp — APPROVED** [review]

**Pull request merged**